checker/cmd/starttls-check/cmd.go

```diff
@@ -1,64 +1,86 @@
 package main
 
 import (
+	"bufio"
+	"encoding/csv"
 	"encoding/json"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"io"
+	"log"
+	"net/http"
 	"os"
-	"strings"
+	"time"
 
 	"github.com/EFForg/starttls-backend/checker"
 )
 
-// Expects domains to be delimited by newlines.
-func domainsFromFile(filename string) ([]string, error) {
-	buff, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return nil, err
+func setFlags() (domain, filePath, url *string) {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
 	}
-	allContent := string(buff)
-	// Filter empty lines from domain list
-	filterDomains := make([]string, 0)
-	for _, line := range strings.Split(allContent, "\n") {
-		trimmed := strings.TrimSpace(line)
-		if len(trimmed) == 0 {
-			continue
-		}
-		filterDomains = append(filterDomains, trimmed)
+	domain = flag.String("domain", "", "Domain to check")
+	filePath = flag.String("file", "", "File path to a CSV of domains to check")
+	url = flag.String("url", "", "URL of a CSV of domains to check")
+
+	flag.Parse()
+	if *domain == "" && *filePath == "" && *url == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
 	}
-	return filterDomains, nil
+	return
 }
 
 // Run a series of security checks on an MTA domain.
 // =================================================
 // Validating (START)TLS configurations for all MX domains.
-//
-// CLI arguments
-// =============
-//     -domain <domain> The domain to perform checks against.
-//
 func main() {
-	// 1. Setup and parse arguments.
-	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "\nNOTE: All checks are enabled by default. "+
-			"Setting any individual 'enable check' flag will disable "+
-			"all checks other than the ones explicitly specified.\n\n")
-		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
-		flag.PrintDefaults()
+	domain, filePath, url := setFlags()
+
+	c := checker.Checker{
+		Cache: checker.MakeSimpleCache(10 * time.Minute),
```

Comment:
Let's make the cache timeout a bit longer since we're doing so many successive checks! Maybe an hour?

Reply:
We can get rid of the cache in the next PR that swaps the hostname check for a noop :)